components/ConditionalIf.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import TruncateMarkup, { TruncateProps } from 'react-truncate-markup';
+import { Modal } from 'react-bootstrap';
+import { commonStyles } from '@htan/data-portal-commons';
+
+interface ConditionalIfValuesProps {
+    attribute: string;
+    attributes: string[];
+    truncateProps?: TruncateProps;
+}
+
+interface ViewAllValuesModalProps {
+    attribute: string;
+    options: JSX.Element[];
+    show: boolean;
+    onClose: () => void;
+}
+
+const ViewAllValuesModal: React.FunctionComponent<ViewAllValuesModalProps> = (
+    props
+) => {
+    return (
+        <Modal show={props.show} onHide={props.onClose}>
+            <Modal.Header closeButton>
+                <Modal.Title>{props.attribute} required if</Modal.Title>
+            </Modal.Header>
+            <Modal.Body>{props.options}</Modal.Body>
+        </Modal>
+    );
+};
+
+const ConditionalIfValues: React.FunctionComponent<ConditionalIfValuesProps> = (
```

Comment:
This is mostly duplicate of [ValidValues](https://github.com/ncihtan/htan-portal/blob/7658f207f7ff991d56333964444e513e3986a698/components/ValidValues.tsx).

Let's generalize and reuse the `ValidValues` component instead. We can name it `TruncatedValuesList` or something like that.